.github/workflows/testing.yml

```diff
@@ -0,0 +1,46 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Testing
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.53'
+          extended: true
+      - name: Build Page
+        run: hugo -v
+      - name: Cache HTMLTest packages
+        uses: actions/cache@v3
+        with:
+          path: |
+            tmp/.htmltest
+          key: ${{ runner.os }}-htmltest
+      - name: Run htmltest
+        uses: wjdp/htmltest-action@master
+        with:
+          path: public
+          config: .htmltest.yml
```

Comment:
Would it make sense to also use the makefile here?

Reply:
here i have to confess, that my GitHub Action knowledge is quiet limited, coming from GitLab CI - and i am not sure what i do have available tooling wise. At Least on GitLab CI it meant to have a container with make and docker running. But i will do some investigations/prototyping, i definitely do see the benefit of having those two using one source, instead of maintaining them twice.

Reply:
we can work on that together at a later stage ;) It's rather simple though, as you can just run commands in GH actions, e.g., 
```yaml

- name: do Something
  run: |
    make build
```

I think docker and make are available out of the box